account_invoice_section_picking/tests/test_account_invoice_section_picking_backorder.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+import datetime
+import logging
+import re
+from types import MethodType
+
+from psycopg2 import sql
+from psycopg2.extensions import AsIs
+
+from .common import TestAccountInvoiceSectionPickingCommon
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from freezegun import freeze_time
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+
+
+class TestAccountInvoiceSectionPickingBackorder(TestAccountInvoiceSectionPickingCommon):
+    def setUp(self):
+        super().setUp()
+        self._monkey_patch_cursor_freeze_time()
+        self.addCleanup(self._reset_monkey_patch_cursor_freezetime)
+
+    def _reset_monkey_patch_cursor_freezetime(self):
+        self.env.cr.execute = self.env.cr.standard_exec
+
+    def _monkey_patch_cursor_freeze_time(self):
+        """Monkey patch cursor to replace SQL TIME/DATE keywords"""
+        # TODO: Extract this part into a fork of freezegun for odoo?
```

Comment:
I'd move it to its own package and provide a decorator and a ctx manager to be used alone. Then we put it in a new repo like `psql-freeze-time` or alike in the OCA org.

Reply:
I'll try to do this as soon as I have time, but I guess we'll have to do something like `odoo-freezegun` because I'm not sure we can do it in psycopg2, since we need it only when using a test cursor.